src/lib/components/custom/dropdown.svelte

```diff
@@ -1,11 +1,16 @@
 <script lang="ts">
+	import type { Snippet } from 'svelte'
+	import Button from '$lib/components/ui/button.svelte'
+
 	type Props = {
 		disabled?: boolean
 		up?: boolean
 		left?: boolean
+		button: Snippet
+		children: Snippet
```

Comment:
Shouldn't the `children` be optional? I mean it works in TS, but not providing children may fail in JS
```suggestion
		children: Snippet
```

Reply:
By this logic everything should be optional. We can do that, however I wanted to indicate here for Typescript users that the children is not optional, because without that the dropdown is just a sticky button, implemented in a convoluted way :)